File: hello-java/audio-stream-server/src/main/java/org/feuyeux/mmap/audio/server/handler/MessageType.java
package org.feuyeux.mmap.audio.server.handler;

/**
 * WebSocket message types enum.
 * All type values are uppercase as per protocol specification.
 */
public enum MessageType {
    START,
    STARTED,
    STOP,
    STOPPED,
    GET,
    ERROR,
    CONNECTED;

    /**
     * Get the uppercase string value for this enum.
     *
     * @return uppercase string representation
     */
    public String getValue() {
        return name();
    }

    /**
     * Parse string to MessageType enum.
     * Case-insensitive comparison for backward compatibility.
     *
     * @param value string value to parse
     * @return corresponding MessageType, or null if not found
     */
    public static MessageType fromString(String value) {
        if (value == null) {
            return null;
        }
        try {
            return MessageType.valueOf(value.toUpperCase());
        } catch (IllegalArgumentException e) {
            return null;
        }
    }
}

File: hello-nodejs/src/server/handler/MessageType.js
/**
 * WebSocket message types enum.
 * All type values are uppercase as per protocol specification.
 */
export class MessageType {
  static START = "START";
  static STARTED = "STARTED";
  static STOP = "STOP";
  static STOPPED = "STOPPED";
  static GET = "GET";
  static ERROR = "ERROR";
  static CONNECTED = "CONNECTED";

  /**
   * Parse string to MessageType.
   * Case-insensitive comparison for backward compatibility.
   *
   * @param {string} value - string value to parse
   * @returns {string|null} corresponding MessageType, or null if not found
   */
  static fromString(value) {
    if (!value) {
      return null;
    }
    const upperValue = value.toUpperCase();
    const validTypes = [
      this.START,
      this.STARTED,
      this.STOP,
      this.STOPPED,
      this.GET,
      this.ERROR,
      this.CONNECTED,
    ];
    return validTypes.includes(upperValue) ? upperValue : null;
  }

  /**
   * Get all valid message types.
   *
   * @returns {string[]} array of all message types
   */
  static values() {
    return [
      this.START,
      this.STARTED,
      this.STOP,
      this.STOPPED,
      this.GET,
      this.ERROR,
      this.CONNECTED,
    ];
  }
}
